### DISCUSSION TITLE
No error reason on websocket disconnect

### DISCUSSION THREAD

**Discussion by yasersakkaf** [fastapi/fastapi#10472] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import json
import time
import traceback

import uvicorn
from fastapi import FastAPI, WebSocket
from fastapi.responses import HTMLResponse
from starlette.websockets import WebSocketDisconnect
from websockets.exceptions import ConnectionClosed

app = FastAPI()

html = """
<!DOCTYPE html>
<html>
    <head>
        <title>Chat</title>
    </head>
    <body>
        <h1>WebSocket Chat</h1>
        <form action="" onsubmit="sendMessage(event)">
            <input type="text" id="messageText" autocomplete="off"/>
            <button>Send</button>
        </form>
        <ul id='messages'>
        </ul>
        <script>
            var ws = new WebSocket("ws://localhost:5049/ws");
            ws.onmessage = function(event) {
                var messages = document.getElementById('messages')
                var message = document.createElement('li')
                var content = document.createTextNode(event.data)
                message.appendChild(content)
                messages.appendChild(message)
            };
            function sendMessage(event) {
                var input = document.getElementById("messageText")
                ws.send(input.value)
                input.value = ''
                event.preventDefault()
            }
        </script>
    </body>
</html>
"""


@app.get("/")
async def get():
    return HTMLResponse(html)


@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    await websocket.accept()
    time.sleep(5)
    while True:
        try:
            data = await websocket.receive_text()
            input_json = json.loads(data)  # This line is deliberately written to cause error
            # query_id = input_json.get("query_id")
            # query_text = input_json.get("query_text")
            # if not query_id or not query_text:
            #     await websocket.send_text(f'{{ "error": "Missing query_id or query_text", "complete": {True} }}')
            #     if not websocket:
            #         await websocket.close()
            #     continue
            print(data)
            await websocket.send_text(f"Message text was: {data}")
        except Exception as error:
            print(str(error))
            if isinstance(error, ConnectionClosed):
                print(f"WebSocket ConnectionClosed: {error.code}, {error.reason}")
            if isinstance(error, WebSocketDisconnect):
                print(f"WebSocket WebSocketDisconnect: {error.code}, {error.reason}")
            print(f"Traceback: {traceback.format_exc()}")
            return


if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=5049)
```

### Description

The error.reason is always empty in the logs whne there is a websocket disconnection. This makes it difficult to debug issues especially with error codes like 1006 and 1005 which can be due to various reasons.

### Operating System

Linux

### Operating System Details

ECS applications deployed using Fargate

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.10

**Comment by Kludex** [comment]
Can you provide a complete example that I can run, please?

**Reply by yasersakkaf** [reply]
Updated !!

**Reply by Kludex** [reply]
The fact that you receive `ConnectionClosed` is an issue in the ASGI specification. We are going to deal with it in the next months.

`WebSocketDisconnect` should have a reason in it, isn't it the case?

**Reply by yasersakkaf** [reply]
No, I don't get any error reason for any Exceptions be it `WebSocketDisconnect` or any other.

**Reply by Kludex** [reply, marked as answer]
Right. Well noticed: https://github.com/encode/starlette/pull/2309

**Reply by yasersakkaf** [reply]
OK - I see the changes on starlette. Now all that's left is to support the latest version of `starlette` i.e. `0.31.1` in `fastapi` latest version i.e. `0.104.0`
